plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/clients/admin/SslKafkaAdminFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.kafka.clients.admin;
+
+import io.trino.plugin.kafka.security.KafkaSecurityConfig;
+import io.trino.spi.connector.ConnectorSession;
+
+import javax.inject.Inject;
+
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+public class SslKafkaAdminFactory
+        implements KafkaAdminFactory
+{
+    private final Properties kafkaProperties;
+
+    @Inject
+    public SslKafkaAdminFactory(PlainTextKafkaAdminFactory plaintextAdminFactory, KafkaSecurityConfig securityConfig)
```

Comment:
I don't like we depend on real implementation (`PlainTextKafkaAdminFactory`). We should use `KafkaAdminFactory`. Otherwise it is difficult to extend it in future, like adding authentication methods etc.

Reply:
Hello @kokosing , if I understood correctly your previous comment [here](https://github.com/trinodb/trino/pull/6929#discussion_r588293965), you recommended to use explicitly `PlaintextKafkaAdminFactory` .

Also, if I'm not mistaken, I think it would not work using `KafkaAdminFactory` since it will already be binded to `SslKafkaAdminFactory `.

Reply:
Yes, you are right. I suggested a composition which you did. Thank you. But I didn't mean that is has to be explicit. I mean that `SslKafkaAdminFactory` depends on `KafkaAdminFactory` (interface) that would be bound to `PlaintextKafkaAdminFactory`. But if we ever implement SCRUM or PLAIN authentication then given `KafkaAdminFactory` could be bound to something else. It is open/closed principle.

> Also, if I'm not mistaken, I think it would not work using KafkaAdminFactory since it will already be binded to SslKafkaAdminFactory as a Singleton.

Yes. So you need to use annotation `@ForKafkaSsl`, constructor would look like:
```
    @Inject
    public SslKafkaConsumerFactory(@ForSslKafka KafkaConsumerFactory delegate,...
```

Reply:
Thank you for the detailed explanation.